sqlglot/parser.py

```diff
@@ -1216,7 +1216,24 @@ def _retreat(self, index: int) -> None:
         if index != self._index:
             self._advance(index - self._index)
 
+    def _warn_unsupported(self) -> None:
+        if len(self._tokens) <= 1:
+            return
+
+        # We use _find_sql because self.sql may comprise multiple chunks, and we're only
+        # interested in emitting a warning for the one being currently processed.
+        sql = self._find_sql(self._tokens[0], self._tokens[-1])
+        if sql:
```

Comment:
when would there not be any sql?

Reply:
if `self.sql` is not ever supplied, which today seems like a possibility? E.g. when you supply a token list directly.

Reply:
we can just get rid of the if sql; check

Reply:
done https://github.com/tobymao/sqlglot/pull/2874/commits/abac3f5b943a3201960ca528566c1809bb4809da